BaseApi/V1/Infrastructure/CorrelationMiddleware.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace BaseApi.V1.Infrastructure
+{
+    public class CorrelationMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public CorrelationMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (context.Request.Headers[Constants.CorrelationId].Count == 0)
```

Comment:
It might be better to use this.HttpContext.Request.Headers.TryGetValue and save the correlationId (if it exists) into a variable

Reply:
apologies, I meant to add this comment to the BaseController

Reply:
No worries, did the work on Base Controller instead :)